src/compiler/parser.rs

```diff
@@ -674,7 +1039,10 @@ fn Parser() -> ParserType {
 //     ParserMut.lock().unwrap()
 // }
 
-#[derive(Copy, Clone)]
-enum ParsingContext {
-    SourceElements,
+bitflags! {
+    pub struct ParsingContext: u32 {
```

Comment:
`ParsingContext` is implemented as a "normal" Typescript enum rather than a "bitflag-ish" enum (where the enum values are specified as `1 << ...`) but it seems to be being used like bitflags (by doing `1 << [enum value]` at the point of usage) so it seems like it makes more sense to just use the bitflags pattern (not sure why they aren't doing that in the Typescript version since they use that pattern for a lot of other enums)